### ISSUE TITLE
Autocompletion not triggering for citations

### ISSUE THREAD

**Issue by HughParsonage** [SublimeText/LaTeXTools#713]
I installed LaTeXTools on ST3 a week ago. After installation, whenever I typed `\cite` and friends, I got the pop-up menu automatically, as expected (and appreciated). But it is now not triggering at all (either when I type `\cite` or when I type `C-l,C-f`.

Internal references (via `\ref{}`) do work normally (via a automatic pop-up menu), just not citations. 

I tried reinstalling ST3 and LaTeXTools (today), but the problem remains. My guess is that this is a problem with my `bibliography.bib`, but I haven't added any entries between it working and it failing. How should I troubleshoot this?

The document I'm having this trouble with compiles fine (including the citations), just the editor settings are off.

(If this is not the right place to ask for help, I apologize -- please direct me where to go and close this.)

**Comment by r-stein** [comment]
The last LaTeXTools release uses a parser to parse the bibfile instead of searching for entries. It might be a good idea to show an error message instead of failing silently.

Could you open the ST console ``ctrl+` `` and show the error message, that occurs when you press `C-l,C-f`.

**Comment by HughParsonage** [comment]
So, I called that routine and got no error message. Now my `\cite` commands are working as normal!

**Comment by grattaninstitute** [comment]
```
Dispatching cite
54763
TEX root: 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\my_report.tex'
Searching file: 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\my_report.tex'
Searching file: 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\CGT-NG-sections\\preamble.tex'
Searching file: 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\package_cite_input.tex'
Bib files found: 
['C:\\Users\\me\\Documents\\GitHub\\my-repo\\bibliography.bib', 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\bibliography2.bib']
Files:
['C:\\Users\\me\\Documents\\GitHub\\my-repo\\bibliography.bib', 'C:\\Users\\me\\Documents\\GitHub\\my-repo\\my-dir\\my-dir-2\\bibliography2.bib']
Loaded 1170 bibitems
Found 1170 total bib entries
Loaded 1200 bibitems
Found 1200 total bib entries
Traceback (most recent call last):
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 465, in __getitem__
    return self._entry[key]
KeyError: 'author_short'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 574, in run_
    return self.run(edit)
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 827, in run
    for completion in completions], on_done)
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 827, in <listcomp>
    for completion in completions], on_done)
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 826, in <listcomp>
    view.window().show_quick_panel([[formatter.vformat(s, (), completion) for s in cite_panel_format] \
  File "./string.py", line 168, in vformat
  File "./string.py", line 190, in _vformat
  File "./string.py", line 253, in get_field
  File "./string.py", line 210, in get_value
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 477, in __getitem__
    return get_author_short(self._entry['author'])
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 442, in get_author_short
    authors = [a.split(" ")[-1] if a[-1] != '}' or a.find('{') == -1 else re.sub(r'{|}', '', a[len(a) - a[::-1].index('{'):-1]) for a in authors]
  File "C:\Users\me\AppData\Roaming\Sublime Text 3\Packages\LaTeXTools\latex_cite_completions.py", line 442, in <listcomp>
    authors = [a.split(" ")[-1] if a[-1] != '}' or a.find('{') == -1 else re.sub(r'{|}', '', a[len(a) - a[::-1].index('{'):-1]) for a in authors]
IndexError: string index out of range
```

I'm not fluent in python, but I understand that this error occurs when `authors` is empty. My `bibliography.bib` doesn't appear to suffer from that problem, however.

**Comment by r-stein** [comment]
Opening the console is not meant as a bugfix and does not solve a problem. I assume you have in your bibliography something like `Author = {Name, Forename and }`. I.e. an `and`, which is not followed by an author name.
I will create a PR to make the name parsing more robust if the format is not `Name1, Forename1 and Name2, Forename2`.

**Comment by grattaninstitute** [comment]
You were right: I did have such an author. 

```
bibliography.raw <- 
  readLines("./bibliography.bib")

authors_with_and_last <- 
  bibliography.raw[grepl("^\\s*(author).*\\s(and)\\s*}", bibliography.raw)]
```

**Comment by ig0774** [comment]
I believe this was resolved when #715 was merged.
